src/gf.c

```diff
@@ -609,6 +615,19 @@ static jl_lambda_info_t *cache_method(jl_methtable_t *mt, union jl_typemap_t *ca
         // in the method cache to prevent anything else from matching this entry
         origtype = NULL;
     }
+    if (makesimplesig && origtype == NULL) {
+        // reduce the complexity of rejecting this entry in the cache
```

Comment:
this ends up in the simplified-signature field. the primary sig field still gets `::Function` (which is also what we bother to print)